src/main/java/com/snowflake/kafka/connector/internal/SnowflakeSinkServiceV1.java

```diff
@@ -87,6 +99,75 @@ public void insert(final Collection<SinkRecord> records) {
     }
   }
 
+  /**
+   * Returns true if we want to skip this record since the value is null or it is an empty json
+   * string.
+   *
+   * <p>Remember, we need to check what is the value schema. Depending on the value schema, we need
+   * to find out if the value is null or empty JSON. It can be empty JSON string in case of custom
+   * snowflake converters.
+   *
+   * <p>If the value is an empty JSON node, we could assume the value passed was null.
+   *
+   * @see com.snowflake.kafka.connector.records.SnowflakeJsonConverter#toConnectData when bytes ==
+   *     null case
+   * @param record record sent from Kafka to KC
+   * @return true if we would skip adding it to buffer -> skip to internal stage and hence skipped
+   *     inside SF Table
+   */
+  private boolean shouldSkipNullValue(SinkRecord record) {
+    boolean isRecordValueNull = false;
+    // get valueSchema
+    Schema valueSchema = record.valueSchema();
+    if (valueSchema instanceof SnowflakeJsonSchema) {
+      // we can conclude this is a custom/KC defined converter.
+      // i.e one of SFJson, SFAvro and SFAvroWithSchemaRegistry Converter
+      if (record.value() instanceof SnowflakeRecordContent) {
+        SnowflakeRecordContent recordValueContent = (SnowflakeRecordContent) record.value();
+        if (recordValueContent.isRecordContentValueNull()) {
+          logDebug(
+              "Record value schema is:{} and value is Empty Json Node for topic {}, partition {}"
+                  + " and offset {}",
+              valueSchema.getClass().getName(),
+              record.topic(),
+              record.kafkaPartition(),
+              record.kafkaOffset());
+          isRecordValueNull = true;
+        }
+      }
+    } else {
+      // Else, it is one of the community converters.
+      // Tombstone handler SMT can be used but we need to check here if value is null if SMT is not
+      // used
+      if (record.value() == null) {
```

Comment:
Umm, no.. Record.key are present for tombstone records since they help identify what key needs to be removed from the corresponding table. (See Tombstone records: https://rmoff.net/2020/11/03/kafka-connect-ksqldb-and-kafka-tombstone-messages/)
(Since we dont have concept of upserts.) Our Connector is append only.